File: ClientApp/src/_models/UserProfileViewModel.ts
import { UserViewModel } from './UserViewModel';

export interface UserProfileViewModel {
    userName: string;
    profileImage: string;
    registrationDate: Date | string;
    isOwnProfile: boolean;
    isFollowing: boolean;
    followers: FollowerViewModel[];
    following: NetworkUserViewModel[];
}

export interface NetworkUserViewModel {
    userName: string;
    profileImage: string;
    isFollowing: boolean;
    isOwnProfile: boolean;
}

export interface FollowerViewModel {
    userName: string;
    profileImage: string;
    isFollowing: boolean;
    isOwnProfile: boolean;
}

File: ViewModels/UserProfileViewModel.cs
using System;
using System.Collections.Generic;

namespace Birder.ViewModels
{
    public class UserProfileViewModel
    {
        public string UserName { get; set; }
        public string ProfileImage { get; set; }
        public DateTime RegistrationDate { get; set; }
        public bool IsOwnProfile { get; set; }
        public bool IsFollowing { get; set; }
        public IEnumerable<FollowerViewModel> Followers { get; set; }
        public IEnumerable<FollowingViewModel> Following { get; set; }
    }
}
